### PR TITLE
settings: Refactor appstore and user management code

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Some refactoring of code for user management and the appstore just to get the changes for my app discover PR small.
This refactors the code to use initial state instead of the custom solution.

image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/lib/Controller/UsersController.php [hunk 1]
```diff
@@ -231,8 +185,10 @@ public function usersList(): TemplateResponse {
 						$userCount -= 1; // we also lower from one the total count
 					}
 				}
-				$userCount += $this->userManager->countUsersOfGroups($groupsInfo->getGroups());
-				$disabledUsers = $this->userManager->countDisabledUsersOfGroups($groupsNames);
+				/** @var UserManager */
+				$userManager = &$this->userManager;
+				$userCount += $userManager->countUsersOfGroups($groupsInfo->getGroups());
+				$disabledUsers = $userManager->countDisabledUsersOfGroups($groupsNames);
```
Comment (Pytal) [inline apps/settings/lib/Controller/UsersController.php:191]: Any reason to not call on `$this->userManager` directly?

Comment (susnux) [reply apps/settings/lib/Controller/UsersController.php]: resolved - needed to change the typing

File: apps/settings/templates/settings-vue.php [hunk 2]
```diff
@@ -1,32 +0,0 @@
...
- *
- */
-
-script('settings', 'vue-settings-apps-users-management');
-style('settings', 'settings');
-
-// Do we have some data to inject ?
-if (is_array($_['serverData'])) {
-	?>
-<span id="serverData" data-server="<?php p(json_encode($_['serverData'])); ?>"></span>
```
Comment (Pytal) [inline apps/settings/templates/settings-vue.php:30]: Let's remove https://github.com/nextcloud/server/blob/84693f89f58a41cdf1964c5bb0558ee6b1e7f1a6/apps/settings/src/App.vue#L30-L36 while we're at it

Comment (susnux) [reply apps/settings/templates/settings-vue.php:30]: Good point - this was fixed in my all-in-one branch but forgot to cherry pick that too.
(Currently have a branch with a bunch of refactoring I did to make the app discover feature implementation easier).

**Review by Pytal — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by nfebe — APPROVED** [review]

**Pull request merged**